backend/src/sms/middlewares/sms.middleware.ts

```diff
@@ -153,22 +145,58 @@ export const InitSmsMiddleware = (
     const logMeta = {
       campaignId,
       label,
-      action: 'getCredentialsFromLabel',
+      action: 'getCredentialsFromLabelCampaign',
     }
     try {
       /* Determine if credential name can be used */
       const campaign = campaignId
         ? await SmsService.findCampaign(+campaignId, userId)
         : null
-      const credentialName = campaign?.demoMessageLimit
+      const canUseDemoCredentials = !!campaign?.demoMessageLimit // false if (1) campaign is null or (2) demoMessageLimit is 0
+      if (!canUseDemoCredentials && label === DefaultCredentialName.SMS) {
+        throw new Error(
+          `Campaign cannot use demo credentials. ${label} is not allowed.`
+        )
+      }
+      const credentialName = canUseDemoCredentials
         ? getDemoCredentialName(label)
         : await getCredentialName(label, +userId)
 
       /* Get credential from the name */
-      const credentials = await credentialService.getTwilioCredentials(
+      res.locals.credentials = await credentialService.getTwilioCredentials(
+        credentialName
+      )
+      res.locals.credentialName = credentialName
+      return next()
+    } catch (err) {
+      const errAsError = err as Error
+      logger.error({
+        ...logMeta,
+        message: `${errAsError.stack}`,
+      })
+      return res.status(400).json({ message: `${errAsError.message}` })
```

Comment:
We shouldn't just return the error message out plainly just in case it contain confidential information. For HTTP errors to be returned with specific message, the try/catch should be very specific to the block containing error that u're tryna catch too